sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/backend/rust/ToRustInnerVisitor.java

```diff
@@ -458,9 +458,18 @@ public VisitDecision preorder(DBSPDecimalLiteral literal) {
         if (type.mayBeNull)
             this.builder.append("Some(");
         String value = Objects.requireNonNull(literal.value).toPlainString();
-        this.builder.append("Decimal::from_str(\"")
+
+        int scale = (type.precision == DBSPTypeDecimal.MAX_PRECISION && type.scale == 0) ? DBSPTypeDecimal.MAX_SCALE : type.scale;
+
+        this.builder.append("SQLDecimal::from_str_with_precision_scale(\"")
                 .append(value)
-                .append("\").unwrap()");
+                .append("\"")
+                .append(", ")
+                .append(type.precision)
```

Comment:
you will have to use the precision constructor if you change the API. That will be more robust instead of appending a suffix.